### PR TITLE
[FEAT] usdt support

### PR DESCRIPTION
**Pull request by gabrielbazan7**

### CODE REVIEW

**Review by rastajpa — DISMISSED** [review]
ACK, nice work

**Review by cmgustavo — APPROVED** [review]
LGTM! 🚀  Tested receiving and sending USDT.

File: src/store/bitpay-id/bitpay-id.effects.ts [hunk 1]
```diff
@@ -462,6 +462,42 @@ const startPairAndLoadUser =
       dispatch(LogActions.error('An error occurred while fetching user data.'));
       dispatch(LogActions.error(errMsg));
     }
+
+    try {
```
Comment (cmgustavo) [inline src/store/bitpay-id/bitpay-id.effects.ts:466]: Duplicated code. Maybe create a new function could be more errorless.

File: src/store/bitpay-id/bitpay-id.effects.ts [hunk 2]
```diff
@@ -524,6 +560,26 @@ export const startFetchDoshToken = (): Effect => async (dispatch, getState) => {
...
+
+      dispatch(
+        BitPayIdActions.successFetchSupportedCurrencies(
+          APP.network,
+          data.user.currencies,
+        ),
+      );
+    } catch (err) {
+      dispatch(LogActions.error('Failed to fetch supported currencies'));
+      dispatch(LogActions.error(JSON.stringify(err)));
```
Comment (cmgustavo) [inline src/store/bitpay-id/bitpay-id.effects.ts:578]: We should handle this kind of errors as `Error` type instead to assume it's a string. 

```
if (err instanceof Error) {
         errMsg = `${err.name}: ${err.message}`;
       } else {
         errMsg = JSON.stringify(err);
       }
```

**Comment by cmgustavo** [comment]
Needs rebase.

**Review by cmgustavo — APPROVED** [review]
LGTM

**Review by Gamboster — APPROVED** [review]
ACK

**Pull request closed without merging**